devito/types/grid.py

```diff
@@ -690,10 +811,9 @@ class SubDomainSet(MultiSubDomain):
     """
 
     def __init__(self, **kwargs):
-        super().__init__()
-
         self._n_domains = kwargs.get('N', 1)
         self._global_bounds = kwargs.get('bounds', None)
+        super().__init__(**kwargs)
```

Comment:
why is that change?

Reply:
Because now `SubDomain`s and `SubDomainSet`s can be passed a `Grid` at instantiation, triggering finalization immediately. For a `SubDomainSet`, `self._n_domains` and `self._global_bounds` are required for finalization.